doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+    ...
+
+However this is of limited use because GHC will only be able to invoke tools
+that are bundled in the installation. It won't be able to call things like
+``make`` or ``autoconf`` which are usually needed for configuring packages, and
+it will also not be able to see any third-party libraries that are installed in
+the system.
+
+.. _Using system libraries with GHC:
+
+Using system libraries with GHC
+-------------------------------
+
+Packages can be installed in an MSYS2 environment via the ``pacman`` package
+manager. The installation of those packages is done in standard directories in
+the system, namely:
+
+- ``C:\msys64\usr\{bin,include,lib}``: for basic POSIX-like utilities. We might
+  use tools that are in ``C:\msys64\usr\bin`` but we should not link against
+  libraries in ``C:\msys64\usr\lib`` as those target ``msys-2.0.dll`` as a
+  runtime dependency which is a fork of Cygwin and not a native Windows DLL.
+  These are referred to as belonging to the ``msys`` repository.
+
+  In order to use tools from ``C:\msys64\usr\bin`` we need to make sure it is
+  present in the ``PATH``.
+
+- ``C:\msys64\<environment>\{bin,include,lib}``: for tools and libraries that
+  are compiled with the appropriate toolchain for that environment. We should
+  provide these directories for includes and library paths if we want to use
+  packages installed in the system.
+
+As an example, see that MSYS2 Packages offer two different versions of
+``openssl``, one for `msys <https://packages.msys2.org/base/openssl>`_ and one
+for `every other environment
+<https://packages.msys2.org/base/mingw-w64-openssl>`_ (and in particular for
+CLANG64 or UCRT64). When installing packages from MSYS2, one should pick the
+options that show ``ucrt64/`` or ``clang64/`` as prefixes, and avoid the ones
+that show ``msys/``.
+
+::
+
+   msys2> pacman -Ss openssl
+   ucrt64/mingw-w64-ucrt-x86_64-openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security (mingw-w64)
+   ...
+   clang64/mingw-w64-clang-x86_64-openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security (mingw-w64)
+   ...
+   msys/openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security
+
+   msys2> pacman -S ucrt64/mingw-w64-ucrt-x86_64-openssl
+
+Once the package is installed, the appropriate location of the libraries should
+be given to GHC, for example if compiling something that depends on
+``libssl.a``:
+
+::
+
+   > ghc ... -IC:\\msys64\\ucrt64\\includes -LC:\\msys64\\ucrt64\\lib -lssl
+
+This should result in a succesful compilation.
+
+Using UNIX-like binaries with GHC
+---------------------------------
+
+As said above, the binaries that we might need to use are (by default) located
+in two places: ``C:\msys64\usr\bin`` and ``C:\msys64\<environment>\bin``. The
+easiest way to give GHC access to these executables is by including those
+directories in your ``PATH``.

```

Comment:
Again: no, not recommended.

Reply:
To me "include in your PATH" does not necessarily mean "set your PATH variable to include this". But I see this is causing a lot of confusion so I left this out.